code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -1535,10 +1537,13 @@
 	name = "Quantum Hair Dye"
 	description = "Has a high chance of making you look like a mad scientist."
 	reagent_state = LIQUID
-	color = "#C8A5DC"
+	color = "#C8A5DC" //gets overwritten
 	var/list/potential_colors = list("0ad","a0f","f73","d14","d14","0b5","0ad","f73","fc2","084","05e","d22","fa0") // fucking hair code
 	taste_description = "sourness"
 
+/datum/reagent/colorful_reagent/Initialize()
+	color = pick(random_color_list)
```

Comment:
I think this is supposed to be on hair_dye, not colorful_reagent

Reply:
Nah, it's randomized.